### PR TITLE
Add support for Razer BlackWidow Tournament Edition 2014

### LINKED ISSUE #1050: Support for BlackWidow Tournament 2014
### Device Info

* Name: BlackWidow Tournament 2014
* Product Number: RZ03-0081
* Model Number: RZ03-00810900-R3M1

### Device Information
```
Bus 003 Device 002: ID 1532:011c Razer USA, Ltd 
Couldn't open device, some information will be missing
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 (Defined at Interface level)
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0         8
  idVendor           0x1532 Razer USA, Ltd
  idProduct          0x011c 
  bcdDevice            2.00
  iManufacturer           1 
  iProduct                2 
  iSerial                 0 
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength           84
    bNumInterfaces          3
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0xa0
      (Bus Powered)
      Remote Wakeup
    MaxPower              100mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      61
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 No Subclass
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     159
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x82  EP 2 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0010  1x 16 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        2
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 No Subclass
      bInterfaceProtocol      2 Mouse
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      94
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               8
```

```
total 0
drwxr-xr-x 2 root root 280 mar 19 12:13 .
drwxr-xr-x 4 root root 660 mar 19 12:13 ..
lrwxrwxrwx 1 root root  10 mar 19 10:58 usb-18f8_USB_OPTICAL_MOUSE-event-if01 -> ../event10
lrwxrwxrwx 1 root root   9 mar 19 10:58 usb-18f8_USB_OPTICAL_MOUSE-event-mouse -> ../event8
lrwxrwxrwx 1 root root   9 mar 19 10:58 usb-18f8_USB_OPTICAL_MOUSE-if01-event-kbd -> ../event9
lrwxrwxrwx 1 root root   9 mar 19 10:58 usb-18f8_USB_OPTICAL_MOUSE-mouse -> ../mouse1
lrwxrwxrwx 1 root root   9 mar 19 12:13 usb-Razer_Razer_BlackWidow_Tournament_Edition-event-if01 -> ../event5
lrwxrwxrwx 1 root root   9 mar 19 12:13 usb-Razer_Razer_BlackWidow_Tournament_Edition-event-kbd -> ../event2
lrwxrwxrwx 1 root root   9 mar 19 12:13 usb-Razer_Razer_BlackWidow_Tournament_Edition-if01-event-kbd -> ../event3
lrwxrwxrwx 1 root root   9 mar 19 12:13 usb-Razer_Razer_BlackWidow_Tournament_Edition-if02-event-mouse -> ../event7
lrwxrwxrwx 1 root root   9 mar 19 12:13 usb-Razer_Razer_BlackWidow_Tournament_Edition-if02-mouse -> ../mouse0
lrwxrwxrwx 1 root root  10 mar 19 10:58 usb-Trust_Webcam_Trust_Webcam_Trust_Webcam-event-if00 -> ../event22
lrwxrwxrwx 1 root root  10 mar 19 10:58 usb-Wacom_Co._Ltd._MTE-450-event-mouse -> ../event11
lrwxrwxrwx 1 root root   6 mar 19 10:58 usb-Wacom_Co._Ltd._MTE-450-mouse -> ../js0
```

### Packet Captures

<Please see here as we probably need packet captures>
<https://github.com/openrazer/openrazer/wiki/Reverse-Engineering-USB-Protocol>
Currently unreachable because I have no way to do a VM to run WireShark.

### Synapse Screenshots
<Screenshots of all available options as seen on Razer Synapse>
Currently unreachable because I have no way to do a VM to run Synapse.

### PR DESCRIPTION
**Pull request by z3ntu**
Fixes #1050

Building from source instructions can be found here: https://github.com/openrazer/openrazer/wiki/Building-a-package

@blshkv @sech1p

### CODE REVIEW

File: daemon/openrazer_daemon/hardware/keyboards.py [hunk 1]
```diff
@@ -405,6 +405,25 @@ class RazerBlackWidowUltimate2013(_MacroKeyboard):
     }
 
 
+class RazerBlackWidowTournamentEdition2014(_MacroKeyboard):
+    """
+    Class for the Razer BlackWidow Tournament Edition 2014
+    """
+    EVENT_FILE_REGEX = re.compile(r'.*Razer_Razer_BlackWidow_Tournament_Edition-(-if01)?-event-kbd')
```
Comment (blshkv) [inline daemon/openrazer_daemon/hardware/keyboards.py]: There is a typo here (extra -), the device is called " /dev/input/by-id/usb-Razer_Razer_BlackWidow_Tournament_Edition-if01-event-kbd -> ../event19"

so it should be Tournament_Edition(-if01)

Comment (z3ntu) [reply daemon/openrazer_daemon/hardware/keyboards.py]: Sorry for the delay, I've fixed it. Could you please test with the most recent version again?

**Comment by z3ntu** [comment]
@blshkv @sech1p Could you please test with the most recent version again?

**Comment by cadorn** [comment]
I am wondering when this is expected to land?

**Comment by z3ntu** [comment]
@cadorn When someone tests this and confirms that everything works as expected, see https://github.com/openrazer/openrazer/wiki/Building-a-package

**Comment by blshkv** [comment]
sorry guys, my keyboard TE 2014 failed after all these years and I decided to purchase a new optical for a change, and it is already supported by openrazer. Won't be testing TE 2014 anymore.

**Comment by bhargu** [comment]
> @cadorn When someone tests this and confirms that everything works as expected, see https://github.com/openrazer/openrazer/wiki/Building-a-package

TLDR; Everything is working with the original BlackWidow Tournament Stealth.

I have tested the commit using a Razer BlackWidow Tournament Edition Stealth (1st edition) which has the same USB device ID (1532:011c). The results are as follows.

With RazerGenie, the illumination control for the logo available. The static and pulsate (breathing) modes work as expected and dimming control is also working.

With Polychromatic, the logo illumination control is working, but the static mode is not available. Only the pulsate mode is available. The Gaming Mode key and Macro Mode key are working too.

The function key is working too, ie. the Fn+\<key\> combos are working.

**Comment by z3ntu** [comment]
@bhargu Great, thanks for testing! Can you please also check if this picture https://assets.razerzone.com/eeimages/products/17564/razer-blackwidow-te-stealth-hero-01.png matches your keyboard?

I'll merge this soon then :)

**Comment by bhargu** [comment]
> @bhargu Great, thanks for testing! Can you please also check if this picture https://assets.razerzone.com/eeimages/products/17564/razer-blackwidow-te-stealth-hero-01.png matches your keyboard?
> 
> I'll merge this soon then :)

Yes, that is the keyboard that I used to test the commit.  As I said, all the features are working. I couldn't test the macro function as I think openrazer currently supports only M1-M5 keys for macro.

**Comment by z3ntu** [comment]
Thanks for testing!
For macros we recommend https://github.com/sezanzeb/key-mapper over the built-in solution. The macro functionality in openrazer will likely be removed at some point in favor of external tools.
